### PR TITLE
Drop CUDA 12.2 Docker images

### PR DESCRIPTION
**Pull request by bdice**
Following https://github.com/rapidsai/docs/pull/526, we can remove CUDA 12.2 from the RAPIDS 24.10 Docker images.

### CODE REVIEW

**Review by jameslamb — APPROVED** [review]
Awesome, thanks for getting that deprecation notice into the 24.08 release.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Bradley and James! 🙏

**Comment by jameslamb** [comment]
I just noticed that tests are not running here on PRs any more.

https://github.com/rapidsai/docker/actions/runs/10392124806?pr=696

Screenshot 2024-08-14 at 2 21 48 PM

I missed that on #702, and `test` should have been added to the checks in #693 😬 

I can put up a PR right now to fix that. It shouldn't block this particular PR though, as this is just deleting things.

**Comment by bdice** [comment]
@jameslamb There is an error in the CI that I am not expecting. Can you take a look? Seems related to the recent CI changes.

**Comment by jameslamb** [comment]
Ok I was confused by the mix of results at first because some jobs were re-run and some weren't. Looking at just the first run, I see what happened:

https://github.com/rapidsai/docker/actions/runs/10477159554/attempts/1?pr=696

image

Some `build` jobs failed, which led to `build-multiarch-manifest` and `test` having the status "skipped"` or something similar, which allowed `delete-temp-images` to run. That deleted some of those images created by `build`, which is why your re-runs of `build-multiarch-manifest` failed.

This is why I want to just remove `delete-temp-images`: https://github.com/rapidsai/docker/pull/708#discussion_r1718850401

This complexity of handling the difference between "failed" and "skipped" is just not worth it here, in my opinion, when we already have another mechanism for cleaning up old stuff.

I'm going to put up a PR proposing that.

For now, for you here, you'll need to do a full re-run (or just wait for that other PR first).

**Comment by jameslamb** [comment]
I've merged latest `branch-24.10` into this, to re-trigger CI and to pull in the changes from #709.

If any builds fail with temporary issues like network errors, just re-running those failed jobs should be safe and successful.

**Comment by jameslamb** [comment]
most build jobs here were failing with what look like temporary errors. I've restarted them.

File: matrix-test.yaml [hunk 1]
```diff
@@ -1,14 +1,13 @@
 # CUDA_VER is `<major>.<minor>` (e.g. `12.0`)
 
 pull-request:
-  - { CUDA_VER: '11.8', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'earliest' }
-  - { CUDA_VER: '12.0', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
-  - { CUDA_VER: '12.2', ARCH: 'arm64', PYTHON_VER: '3.11', GPU: 'a100', DRIVER: 'latest' }
+  - { CUDA_VER: '11.8', ARCH: 'amd64', PYTHON_VER: '3.9', GPU: 'v100', DRIVER: 'earliest' }
```
Comment (jameslamb) [inline matrix-test.yaml:4]:
```suggestion
  - { CUDA_VER: '11.8', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'earliest' }
```

3.9 images were removed in #709: https://github.com/rapidsai/docker/pull/709#issuecomment-2303271547

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**